### PR TITLE
Add SKU as a source field for the barcode.

### PR DESCRIPTION
**Pull request by Ollie222**

### CODE REVIEW

File: modules/label/modules/barcode/commerce_pos_label_barcode.module [hunk 1]
```diff
@@ -71,19 +74,28 @@ function commerce_pos_label_barcode_generate($product, $format) {
...
   $upc_field = variable_get('commerce_pos_label_barcode_upc_field_' . $product->type, FALSE);
 
-  if (!empty($upc_field) && $field_upc = field_get_items('commerce_product', $product, $upc_field)) {
-    try {
-      $barcode_settings = $format['barcode'];
-
-      $barcode =  $generator->getBarcode($field_upc[0]['value'], $barcode_settings['type'], $barcode_settings['widthFactor'], $barcode_settings['totalHeight'], $barcode_settings['color']);
-      return $barcode;
+  if(!empty($upc_field)) {
+    if($upc_field == 'sku') {
```
Comment (travis-bradbury) [inline modules/label/modules/barcode/commerce_pos_label_barcode.module:78]: Would it be nicer to use an `EntityMetadataWrapper` for `$product` so we can do `$product_wrapper->{$upc_field}->value()` without caring what the name of it is?

File: modules/label/modules/barcode/commerce_pos_label_barcode.module [hunk 2]
```diff
@@ -71,19 +74,28 @@ function commerce_pos_label_barcode_generate($product, $format) {
   $generator = new BarcodeGeneratorPNG();
   $upc_field = variable_get('commerce_pos_label_barcode_upc_field_' . $product->type, FALSE);
 
-  if (!empty($upc_field) && $field_upc = field_get_items('commerce_product', $product, $upc_field)) {
-    try {
-      $barcode_settings = $format['barcode'];
-
-      $barcode =  $generator->getBarcode($field_upc[0]['value'], $barcode_settings['type'], $barcode_settings['widthFactor'], $barcode_settings['totalHeight'], $barcode_settings['color']);
-      return $barcode;
+  if(!empty($upc_field)) {
```
Comment (dylf) [inline modules/label/modules/barcode/commerce_pos_label_barcode.module:77]: There should be a space in between your if statement and the opening parentheses.

File: modules/label/modules/barcode/commerce_pos_label_barcode.module [hunk 3]
```diff
@@ -71,19 +74,28 @@ function commerce_pos_label_barcode_generate($product, $format) {
...
+        $upc_field_value = $field_upc[0]['value'];
+      }
     }
-    catch (BarcodeException $e) {
-      watchdog('commerce_pos_label', 'Unable to generate barcode for product @sku. @exception: @message', array(
-        '@exception' => get_class($e),
-        '@sku' => $product->sku,
-        '@message' => $e->getMessage(),
-      ), WATCHDOG_ERROR);
+    if($upc_field_value) {
```
Comment (dylf) [inline modules/label/modules/barcode/commerce_pos_label_barcode.module:86]: Add a space between if & parentheses

**Comment by smmccabe** [comment]
Merged via patch on d.o

**Pull request closed without merging**